backend/danswer/indexing/indexing_pipeline.py

```diff
@@ -103,40 +126,25 @@ def index_doc_batch(
     memory requirements"""
     with Session(get_sqlalchemy_engine()) as db_session:
         document_ids = [document.id for document in documents]
-
-        # Skip indexing docs that don't have a newer updated at
-        # Shortcuts the time-consuming flow on connector index retries
         db_docs = get_documents_by_ids(
             document_ids=document_ids,
             db_session=db_session,
         )
         id_to_db_doc_map = {doc.id: doc for doc in db_docs}
-        id_update_time_map = {
-            doc.id: doc.doc_updated_at for doc in db_docs if doc.doc_updated_at
-        }
-
-        updatable_docs: list[Document] = []
-        if ignore_time_skip:
-            updatable_docs = documents
-        else:
-            for doc in documents:
-                if (
-                    doc.id in id_update_time_map
-                    and doc.doc_updated_at
-                    and doc.doc_updated_at <= id_update_time_map[doc.id]
-                ):
-                    continue
-                updatable_docs.append(doc)
 
+        # Skip indexing docs that don't have a newer updated at
+        # Shortcuts the time-consuming flow on connector index retries
+        updatable_docs = (
+            get_doc_ids_to_update(documents=documents, db_docs=db_docs)
+            if not ignore_time_skip
```

Comment:
ew :( double negatives (just kidding, dont change it)

documents if ignore time skip else get_doc_ids_to_update